src/Avalonia.Base/Animation/Animatable.cs

```diff
@@ -29,6 +29,12 @@ public class Animatable : AvaloniaObject
         private bool _transitionsEnabled = true;
         private bool _isSubscribedToTransitionsCollection = false;
         private Dictionary<ITransition, TransitionState>? _transitionState;
+        readonly NotifyCollectionChangedEventHandler _collectionChanged;
+
+        public Animatable()
+        {
+            _collectionChanged = TransitionsCollectionChanged;
```

Comment:
This is technically a memory regression since not every animatable needs to have transitions. I would suggest an approach like in https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Controls/Shapes/Path.cs#L25.